### PR TITLE
Improved tools (check if the examples are valid)

### PR DESCRIPTION
**Pull request by UtechtDustin**
**Description**
This PR will add a new check to our tool, it checks if the example in our documentation are formatted correctly `[name](url)` and checks if the file exists on our examples repository.

@wenzhixin currently the tool use the github api directly, but the api has a limit of 60 requests per hour which could be exceeded while developing. You could write a script which cache the json(s) on our server (quite simple cronjob).

While developing i also fixed 5 broken examples, there are two more examples which are broken (this is btw also the reason why the ci fails for this PR):

- [html](https://bootstrap-table.com/docs/api/table-options/#html)
- [virtualScrollItemHeight](https://bootstrap-table.com/docs/api/table-options/#virtualscrollitemheight)

i dont know what that options do, maybe one of you could add the examples (or change the url if there is already a example).

**Bug fix?**
no

**New Feature?**
yes

**Resolve an issue?**
no

**Example(s)?**
no

### CODE REVIEW

**Comment by wenzhixin** [comment]
Updated the code, I think the `html` is unnecessary and never used, so I removed it.

**Comment by UtechtDustin** [comment]
@wenzhixin I dont used the filesystem to prevent that every fork/contributor who use the command must have the other repository as well.
Maybe we should check if the repository was found on that location, if not ignore that check and just output a warning that this check was skipped because the repository is missing.

**Comment by wenzhixin** [comment]
OK,  can you help to add a waring?

**Comment by UtechtDustin** [comment]
@wenzhixin sure let me check that late :-)

**Comment by UtechtDustin** [comment]
@wenzhixin Added a warning :-)

File: tools/check-api.js [hunk 1]
```diff
@@ -4,6 +4,20 @@ const fs = require('fs')
...
+let exampleFiles = []
+if (exampleFilesFound) {
+  exampleFiles = [
+    ...fs.readdirSync(exampleFilesFolder + 'welcomes'),
+    ...fs.readdirSync(exampleFilesFolder + 'options'),
+    ...fs.readdirSync(exampleFilesFolder + 'column-options'),
+    ...fs.readdirSync(exampleFilesFolder + 'methods')
+  ]
+} else {
+  console.log((chalk.yellow(chalk.bold('Warning: ') + 'Cant check if example files are correct formatted and has a valid url.')))
```
Comment (djhvscf) [inline tools/check-api.js:18]: have a valid...*

File: tools/check-api.js [hunk 2]
```diff
@@ -4,6 +4,20 @@ const fs = require('fs')
...
+  console.log((chalk.yellow(chalk.bold('Warning: ') + 'To enable that check, please clone the "bootstrap-table-examples" repository in the tools folder or create a symlink (if you already cloned the repository on a other path).')))
```
Comment (djhvscf) [inline tools/check-api.js:19]: an other...*

File: tools/check-api.js [hunk 3]
```diff
@@ -39,7 +54,21 @@ class API {
           if (this.ignore && this.ignore[key] && this.ignore[key].includes(name)) {
             continue
           }
-          if (!details[i + 1] || details[i + 1].indexOf(`**${name}:**`) === -1) {
+
+          const tmpDetails = details[i + 1].trim()
+          if (name === 'Example' && exampleFilesFound) {
+            const matches = exampleRegex.exec(tmpDetails)
+            if (matches === null) {
```
Comment (djhvscf) [inline tools/check-api.js:61]: Could we validate this like a falsy value?

`if (!matches) {....`

**Pull request merged**